File: src/Algorithm/__Challenge__/leetcode.26/removeDuplicates.js
/**
 * @param {number[]} nums
 * @return {number}
 */
function removeDuplicates(nums) {
    if(nums.length == 1) return 1;

    for(let i = 1; i < nums.length; i++){
        if(nums[i] == nums[i-1]){
            let hi = i;
            while (nums[hi] == nums[i] && hi < nums.length - 1) hi++;
            for(let j = i; j < hi; j++) nums[j] = nums[hi];
        }
    }
    return countDuplicates(nums);
};

function countDuplicates(nums){
    let count = 1;
    for(let i = 1; i< nums.length; i++){
        if(nums[i] != nums[i-1]) count++
    }
    return count;
}
console.log(removeDuplicates([1,1,2]))

File: src/Algorithm/__Challenge__/leetcode.80/removeDuplicates.go
package leetcode

func removeDuplicates(nums []int) int {
	var p = 2
	for i := 2; i < len(nums); i++ {
		if nums[p-2] != nums[i] {
			nums[p] = nums[i]
			p++
		}
	}
	return p
}
